doc/sphinx/Pacemaker_Explained/status.rst

```diff
@@ -1,280 +1,347 @@
-   |                  | The name of the source function that made the most       |
-   |                  | recent change (for debugging purposes).                  |
-   +------------------+----------------------------------------------------------+
-   | join             | .. index::                                               |
-   |                  |    single: join; node status                             |
-   |                  |    single: node; status, join                            |
-   |                  |                                                          |
-   |                  | Does the node participate in hosting resources?          |
-   |                  | Allowed values: ``down``, ``pending``, ``member``.       |
-   |                  | ``banned``.                                              |
-   +------------------+----------------------------------------------------------+
-   | expected         | .. index::                                               |
-   |                  |   single: expected; node status                          |
-   |                  |   single: node; status, expected                         |
-   |                  |                                                          |
-   |                  | Expected value for ``join``.                             |
-   +------------------+----------------------------------------------------------+
-   
-The cluster uses these fields to determine whether, at the node level, the
-node is healthy or is in a failed state and needs to be fenced.
-   
+.. list-table:: **Attributes of a node_state Element**
+   :class: longtable
+   :widths: 1 1 3
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Description
+   * - .. _node_state_id:
+
+       .. index::
+          pair: node_state; id
+
+       id
+     - :ref:`text <text>`
+     - Node ID (identical to ``id`` of corresponding ``node`` element in the
+       ``configuration`` section)
+   * - .. node_state_uname:
+
+       .. index::
+          pair: node_state; uname
+
+       uname
+     - :ref:`text <text>`
+     - Node name (identical to ``uname`` of corresponding ``node`` element in the
+       ``configuration`` section)
+   * - .. node_state_in_ccm:
+
+       .. index::
+          pair: node_state; in_ccm
+
+       in_ccm
+     - :ref:`epoch time <epoch_time>` *(since 2.1.7; previously boolean)*
+     - If the node's controller is currently joined at the cluster layer, the
+       epoch time at which it did so (or 1 if the node is in the process of
+       leaving the cluster), otherwise 0 *(since 2.1.7; previously, it was
+       "true" or "false")*
+   * - .. node_state_crmd:
+
+       .. index::
+          pair: node_state; crmd
+
+       crmd
+     - :ref:`epoch time <epoch_time>` *(since 2.1.7; previously an enumeration)*
+     - If the node's controller is currently joined at the cluster messaging
```

Comment:
should be consistent in using "cluster layer" or "cluster messaging layer" in `in_ccm` and `crmd` descriptions